WordPress/gutenberg
# Editor: Improve render performance (scrolling and typing)

<img width="1291" alt="85607658-520dc980-b622-11ea-8a7c-0d80a6ef5e1c" src="https://user-images.githubusercontent.com/2322354/86044359-fffce780-ba17-11ea-949d-fde04930a048.png">

This update improves the rendering performance of the editor, especially for user interactions such as typing and scrolling.

This was achieved using a combination of CSS and JS updates that leverage GPU rendering where appropriate and optimize layout calculations.

Here are the results:

(Recorded on Chrome with the [paint rendering debug tool](https://d.pr/v/uAhLyG) enabled).
A green flash indicates a browser repaint.

**Before**
https://d.pr/v/g3eX8A

Notice that every interaction causes the entire editor to flash. It's bad enough when scrolling is involved. It's particularly concerning when the user types anything (anywhere, including the sidebar).

**After**
https://d.pr/v/I3uOyb

Wayyyy less repainting! Notice when typing, the repaint is contained in the typing area (e.g. RichText or `input`)


## How has this been tested?

This has been tested in local Gutenberg across Chrome, Firefox, and Safari.

* Run `npm install`
* Run `npm run dev`
* Use the editor (add content, etc...)
* Ensure that things render correctly

Certain things to pay attention to. Anything involving popovers (e.g. Toolbar, color picker).

For visual repaint testing...

* Enable Chrome's [paint rendering debug tool](https://d.pr/v/uAhLyG) (optional)
* Use the editor
* Analyze when/where things are repainting


## Types of changes

* Moved the `<Popover.Slot />` for BlockToolbar
* Added CSS layering with `translate3d(0, 0, 0)` for various elements


## Checklist:
- [n/a] My code follows the accessibility standards. 
- [n/a] I've updated all React Native files affected by any refactorings/renamings in this PR. 

Resolves: https://github.com/WordPress/gutenberg/issues/23427
